### PR TITLE
[6.3 Feature] Canned Role Org Admin API stubs

### PR DESCRIPTION
**Pull request by jyejare**
$subject

### CODE REVIEW

File: tests/foreman/api/test_role.py [hunk 1]
```diff
@@ -245,65 +248,751 @@ def test_negative_update_role_taxonomies(self):
...
-            2. Unset the override flag in above role filter
+            1. Create role with organization A and Location A
+            2. Create an overridden filter in role with organization B
+                and Location B
+            3. Set above filter override flag to False in role
+            4. Get above role filters
 
-        :expectedresults: The taxonomies of filters resets/synced to role
-            taxonomies
+        :expectedresults: On unsetting filter override, role filter should
```
Comment (oshtaier) [inline tests/foreman/api/test_role.py]: previous expected result sounds better or it is not valid

File: tests/foreman/api/test_role.py [hunk 2]
```diff
@@ -245,65 +248,751 @@ def test_negative_update_role_taxonomies(self):
...
+    @tier1
+    def test_positive_create_org_admin_from_clone(self):
+        """Create Org Admin role which has access to most of the resources
+        within organization
+
+        :id: bf33b70a-25a9-4eb1-982f-03448d008ec8
+
+        :steps:
+
+            1. Create Org Admin role by cloning 'Organization admin' role which
```
Comment (oshtaier) [inline tests/foreman/api/test_role.py]: as I mentioned in my previous reviews to different PRs, it is not logical to have 1 step which is clone of test case description. I am suggesting to remove `steps` section at all for such situations

File: tests/foreman/api/test_role.py [hunk 3]
```diff
@@ -245,65 +248,751 @@ def test_negative_update_role_taxonomies(self):
...
+        :expectedresults:
+
+            1. While cloning, role allows to set taxonomies
+            2. New taxonomies should be applied to cloned role successfully
+
+        :caseautomation: notautomated
+        """
+
+    @stubbed
+    @tier3
```
Comment (oshtaier) [inline tests/foreman/api/test_role.py]: Only roles and users are affected, so tier2 is more logical here

File: tests/foreman/api/test_role.py [hunk 4]
```diff
@@ -245,65 +248,751 @@ def test_negative_update_role_taxonomies(self):
...
+
+        :id: ab709822-0e58-473b-994c-b4e660d2bf76
+
+        :steps:
+
+            1. Create Org Admin role and assign taxonomies to it
+            2. Create user with same taxonomies as role above
+            3. Assign cloned role to user above
+            4. Login with user and attempt to access resources
+
```
Comment (oshtaier) [inline tests/foreman/api/test_role.py]: please put proper `:CaseLevel:`

File: tests/foreman/api/test_role.py [hunk 5]
```diff
@@ -245,65 +248,751 @@ def test_negative_update_role_taxonomies(self):
...
+        :expectedresults: User should be able to access all the resources and
+            permissions in taxonomies selected in Org Admin role
+        """
+
+    @stubbed
+    @tier3
```
Comment (oshtaier) [inline tests/foreman/api/test_role.py]: same here about tier2

File: tests/foreman/api/test_role.py [hunk 6]
```diff
@@ -245,65 +248,751 @@ def test_negative_update_role_taxonomies(self):
...
+        :id: 225bed99-e768-461f-9972-4b9a4ffe3152
+
+        :steps:
+
+            1. Create Org Admin and assign taxonomies to it
+            2. Create user with different taxonomies than above Org Admin
+                taxonomies
+            3. Assign above cloned role to user
+            4. Login with user and attempt to access resources
+
```
Comment (oshtaier) [inline tests/foreman/api/test_role.py]: please put proper `:CaseLevel:`

File: tests/foreman/api/test_role.py [hunk 7]
```diff
@@ -245,65 +248,751 @@ def test_negative_update_role_taxonomies(self):
...
+        :expectedresults: User should not be able to access any resources and
+            permissions in taxonomies selected in Org Admin role
+        """
+
+    @stubbed
+    @tier3
```
Comment (oshtaier) [inline tests/foreman/api/test_role.py]: same about tier2

File: tests/foreman/api/test_role.py [hunk 8]
```diff
@@ -245,65 +248,751 @@ def test_negative_update_role_taxonomies(self):
...
+        :id: 5d663705-7b9e-4c42-82fa-db2f53715c91
+
+        :steps:
+
+            1. Create Org Admin and assign taxonomies to it
+            2. Create user with different taxonomies than above Org Admin
+                taxonomies
+            3. Assign above cloned role to user
+            4. Login with user and attempt to access resources
+
```
Comment (oshtaier) [inline tests/foreman/api/test_role.py]: please put proper `:CaseLevel:`

File: tests/foreman/api/test_role.py [hunk 9]
```diff
@@ -245,65 +248,751 @@ def test_negative_update_role_taxonomies(self):
...
+            2. Clone above role
+            3. Attempt to override the filter in cloned role
 
-        :expectedresults: User should be able to access the resource(s) of the
-            assigned role
+        :expectedresults: Filter in cloned role should be overriding
 
         :caseautomation: notautomated
 
-        :CaseLevel: Integration
```
Comment (oshtaier) [inline tests/foreman/api/test_role.py]: tier2 so please leave that statement and in next test cases

File: tests/foreman/api/test_role.py [hunk 10]
```diff
@@ -245,65 +248,751 @@ def test_negative_update_role_taxonomies(self):
...
-            2. Create resource(s) filter(s) without overriding them
-            3. Create user with taxonomies same as role taxonomies
-            4. Assign step 1 role to user
+            1. Create a role with overridden filter
+            2. Clone above role
+            3. Attempt to override the filter in cloned role
 
-        :expectedresults: User should be able to access the resource(s) of the
-            assigned role
+        :expectedresults: Filter in cloned role should be overriding
```
Comment (oshtaier) [inline tests/foreman/api/test_role.py]: `should be overridden`

File: tests/foreman/api/test_role.py (hunk 2 above)
Comment (jyejare) [inline tests/foreman/api/test_role.py]: This step describes From which role(Organization admin) org admin role can be created. So this step should be there.

File: tests/foreman/api/test_role.py (hunk 3 above)
Comment (jyejare) [inline tests/foreman/api/test_role.py]: No it is going to access the resources such as domains , repositories as step 4 mentioned. So this is tier3 test.

File: tests/foreman/api/test_role.py (hunk 5 above)
Comment (jyejare) [inline tests/foreman/api/test_role.py]: same comment as above

File: tests/foreman/api/test_role.py (hunk 7 above)
Comment (jyejare) [inline tests/foreman/api/test_role.py]: same comment as above

**Comment by jyejare** [comment]
@oshtaier , I have fixed your comment wherever needed and replied on some. Please have another look.

**Comment by oshtaier** [comment]
Basically, it is huge scope to understand right away, but looks good from first glance

**Comment by jyejare** [comment]
@SatelliteQE/quality-engineers , Guys need one more ACK. Please visit

File: tests/foreman/api/test_role.py [hunk 11]
```diff
@@ -287,23 +391,658 @@ def test_positive_access_resources_from_role_taxonomies(self):
...
+            2. Assign any taxonomies to it
+            3. Create an user and assign above role to user
+            4. Delete the Org Admin role
+
+        :expectedresults: Super Admin should be able to remove Org Admin role
+        """
+
+    @stubbed
+    @tier2
+    def test_positive_taxonomies_control_to_superAdmin_with_org_admin(self):
```
Comment (renzon) [inline tests/foreman/api/test_role.py]: Any special reason for this upper "A" char?

File: tests/foreman/api/test_role.py [hunk 12]
```diff
@@ -287,23 +391,658 @@ def test_positive_access_resources_from_role_taxonomies(self):
...
+
+        :expectedresults: Super admin should be able to access the entities in
+            taxonomies assigned to Org Admin
+
+        :CaseLevel: Integration
+        """
+
+    @stubbed
+    @tier2
+    def test_positive_taxonomies_control_to_superAdmin_without_org_admin(self):
```
Comment (renzon) [inline tests/foreman/api/test_role.py]: Same as above

File: tests/foreman/api/test_role.py (hunk 11 above)
Comment (jyejare) [inline tests/foreman/api/test_role.py]: No Reason, just fixing it.

**Pull request merged**